### PR TITLE
adds frame extraction to add_new_videos

### PR DESCRIPTION
**Pull request by louisadamian**
Adds frame extraction to add_new_videos for **only** the newly added videos. Modifies extract_frames to accept an optional list of strings containing paths, when that list is not empty extract_frames only extracts frames for the videos at those paths.

Currently there is no way to extract frames for only videos added after initial model creation with user input in the command line.

### CODE REVIEW

**Comment by jeylau** [comment]
Thanks @louisadamian for the contribution! I like that behavior of auto-extracting frames upon adding new videos. We may still want to be able to turn it off for the people who extract their frames manually via the GUI, what do you think @MMathisLab?

**Comment by louisadamian** [comment]
I can definitely add an `extract_frames` optional boolean argument

File: deeplabcut/create_project/add.py [hunk 1]
```diff
@@ -99,8 +105,12 @@ def add_new_videos(config, videos, copy_videos=False, coords=None):
...
             cfg["video_sets_original"].update(params)
+    videos_str = [str(video) for video in videos]
+    if extract_frames:
+        frame_extraction.extract_frames(config, userfeedback=False, videos=videos_str)
 
     auxiliaryfunctions.write_config(config, cfg)
+
     print(
-        "New video was added to the project! Use the function 'extract_frames' to select frames for labeling."
+        "New videos were added to the project and frames have been extracted for labeling!"
```
Comment (aristotelis-economides) [inline deeplabcut/create_project/add.py]: You might want to keep the new print statement within the `if` and put the original print statement in an `else` (so the message is correct depending on the bool value)

File: deeplabcut/generate_training_dataset/frame_extraction.py [hunk 2]
```diff
@@ -142,6 +143,9 @@ def extract_frames(
         The index of the camera that already has extracted frames. This will match frame numbers to extract for all other cameras.
         This parameter is necessary if you wish to use epipolar lines in the labeling toolbox. Only use if mode = 'match' and config3d is provided.
 
+    videos: list, default: None
+            A list of the string containing full paths to videos to extract frames for. If this is left as None all videos in the config file will have frames extracted.
```
Comment (aristotelis-economides) [inline deeplabcut/generate_training_dataset/frame_extraction.py]: minor typo here "A list of the string containing..." --> "A list of string**s** containing..."

**Review by aristotelis-economides — CHANGES_REQUESTED** [review]
Nice stuff @louisadamian thanks for the contribution :) I mentioned some minor things

**Comment by louisadamian** [comment]
thanks @intergalactic-mammoth

File: deeplabcut/create_project/add.py (hunk 1 above)
Comment (aristotelis-economides) [reply deeplabcut/create_project/add.py]: Is this fixed? If I'm not mistaken now the function will print `New videos were added to the project and frames have been extracted for labeling` even if `extract_frames` is False.

If it is false it should still print `New videos were added to the project! Use the function 'extract_frames' to select frames for labeling` no?

Or am I missing something?

**Review by aristotelis-economides — CHANGES_REQUESTED** [review]

**Comment by louisadamian** [comment]
@intergalactic-mammoth my bad. It's fixed now.

**Comment by aristotelis-economides** [comment]
> @intergalactic-mammoth my bad. It's fixed now.

Nice! 🙌🔥

Maybe keep this part `Use the function 'extract_frames' to select frames for labeling` in the original print statement so that people can still get directions on what to do next ;)

Otherwise looks great to me!

**Review by aristotelis-economides — APPROVED** [review]

File: deeplabcut/create_project/add.py [hunk 3]
```diff
@@ -112,5 +112,7 @@ def add_new_videos(config, videos, copy_videos=False, coords=None, extract_frame
             "New videos were added to the project and frames have been extracted for labeling!"
         )
     else:
-        print("New videos were added to the project! use Use the function 'extract_frames' to select frames for labeling.")
+        print(
+            "New videos were added to the project! use Use the function 'extract_frames' to select frames for labeling."
```
Comment (aristotelis-economides) [inline deeplabcut/create_project/add.py]: typo -> "use Use"

**Review by AlexEMG — APPROVED** [review]

**Review by jeylau — APPROVED** [review]

**Pull request merged**